### PR TITLE
Vtol weathervane cleaned up history

### PR DESCRIPTION
**Pull request by tumbili**
Fixes https://github.com/PX4/Firmware/issues/3679
@sanderux As discussed I cleaned up the history so that we only remain with three commits.
The diff to the old branch is zero so everything should be inside. Will give this a test in SITL and Ada can maybe test fly it today.
@AndreasAntener  FYI

### CODE REVIEW

**Comment by tumbili** [comment]
Please delete the old branch (referenced above ) if there aren't any objections.

**Comment by sanderux** [comment]
Done, looks good

**Comment by tumbili** [comment]
@sanderux @AndreasAntener This is flight tested in SITL for both landing and loiter modes. Checked the log files, it seems to do the right thing.

**Comment by AndreasAntener** [comment]
Rain just stopped me right after flashing this branch... I hope the weather is good for you Sander tomorrow :)

**Comment by sanderux** [comment]
There is some wind but that would be the point for this pr. Should stay dry

**Comment by tumbili** [comment]
@sanderux @AndreasAntener Parameters to activate this are
VT_WV_LND_EN enable mode for landing waypoint
VT_WV_LTR_EN enable mode for loiter items
VT_WV_YAWR_SCL scale factor for desired yaw_rate setpoint

**Comment by AndreasAntener** [comment]
VT_WV_LTR_EN doesn't seem to be used anywhere

**Comment by tumbili** [comment]
@AndreasAntener https://github.com/PX4/Firmware/blob/vtol_weather_vane/src/modules/navigator/mission_block.cpp#L75

**Comment by sanderux** [comment]
In mission block

**Comment by AndreasAntener** [comment]
Funny, QGC didn't show me this on the field, anyway, it's there :)

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 1]
```diff
@@ -1263,6 +1263,14 @@ MulticopterPositionControl::task_main()
 				control_auto(dt);
 			}
 
+			/* weather-vane mode for vtol: disable yaw control */
+			if(_mode_auto && _pos_sp_triplet.current.disable_mc_yaw_control == true) {
```
Comment (AndreasAntener) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: we should change `_mode_auto` to `!_control_mode.flag_control_manual_enabled` so this also works in offboard and it's consistent with the other checks

**Comment by sanderux** [comment]
Log from weathervane loiter test: http://logs.uaventure.com/view/t7a2AZB3WHgRZsDjGCEJpi

File: src/modules/mc_att_control/mc_att_control_main.cpp [hunk 2]
```diff
@@ -724,6 +734,14 @@ MulticopterAttitudeControl::control_attitude(float dt)
...
+	/* weather-vane mode, dampen yaw rate */
+	if (_v_att_sp.disable_mc_yaw_control == true && _v_control_mode.flag_control_velocity_enabled && !_v_control_mode.flag_control_manual_enabled) {
+		float wv_yaw_rate_max = _params.auto_rate_max(2) * _params.vtol_wv_yaw_rate_scale;
+		_rates_sp(2) = math::constrain(_rates_sp(2), -wv_yaw_rate_max, wv_yaw_rate_max);
+		// prevent integrator winding up in weathervane mode
+		_rates_int(2) = 0.0f;
+	}
+
 	/* feed forward yaw setpoint rate */
 	_rates_sp(2) += _v_att_sp.yaw_sp_move_rate * yaw_w * _params.yaw_ff;
```
Comment (AndreasAntener) [inline src/modules/mc_att_control/mc_att_control_main.cpp]: @tumbili should we also remove the feed forward when we clamp the rate?

**Comment by AndreasAntener** [comment]
This looks good. Let's address my 2 comments. We should check it on landing as well even though I don't think it's problematic.
screen shot 2016-02-14 at 15 07 12

**Comment by AndreasAntener** [comment]
Addressed comment 1

**Comment by sanderux** [comment]
This should address the feed forward

**Comment by tumbili** [comment]
Merged!

**Pull request merged**